File: 2022/2/puzzle2.py
"""provides path joining functions annd module joining"""
import os
import sys

INPUT_FILE="input"


def obj_to_score(player, opponent):
  match player:
    case "X":
      match opponent:
        case "A":
          return 3
        case "B":
          return 1
        case _:
          return 2
    case "Y":
      match opponent:
        case "A":
          return 1
        case "B":
          return 2
        case _:
          return 3
    case _:
      match opponent:
        case "A":
          return 2
        case "B":
          return 3
        case _:
          return 1
def game_score(player):
  match player:
    case "X":
      return 0
    case "Y":
      return 3
    case _:
      return 6


def main():
  """Main Function called on Startup"""
  input_text= open(os.path.join(sys.path[0], INPUT_FILE),'r', encoding="UTF-8")
  lines = input_text.readlines()
  
  total_score = 0
  for line in lines:
    opponent = line.strip().split(' ')[0]
    player = line.strip().split(' ')[1]
    total_score += obj_to_score(player, opponent) + game_score(player)
  print(total_score)

if __name__ == "__main__":
  main()

File: 2022/2/Puzzle2.cs
using adventofcode;

namespace AdventOfCode22.Day02
{
    public class Puzzle2
    {
        public const string INPUTFILE = "inputtest";
        public static void Run()
        {
            string[] lines = File.ReadAllLines(Program.GetInputPath(INPUTFILE, "Day02"));
            int totalScore = 0;
            foreach (string line in lines)
            {
                char opponent = line[0];
                char player = line[2];
                totalScore += ObjToScore(player, opponent) + GameScore(player);
            }
            Console.WriteLine(totalScore);
        }

        private static int GameScore(char player)
        {
            return player switch
            {
                'X' => 0,
                'Y' => 3,
                _ => 6
            };
        }

        private static int ObjToScore(char player, char opponent)
        {
            return player switch
            {
                'X' => opponent switch
                {
                    'A' => 3,
                    'B' => 1,
                    _ => 2,
                },
                'Y' => opponent switch
                {
                    'A' => 1,
                    'B' => 2,
                    _ => 3,
                },
                _ => opponent switch
                {
                    'A' => 2,
                    'B' => 3,
                    _ => 1,
                },
            };
        }
    }
}
